File: clients/java/examples/collections-find-one-and-update-vector.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.Update;
import com.datastax.astra.client.collections.commands.options.CollectionFindOneAndUpdateOptions;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Sort;
import java.util.Optional;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Update a document
    Update update = Update.create().set("color", "blue");
    CollectionFindOneAndUpdateOptions options =
        new CollectionFindOneAndUpdateOptions()
            .sort(Sort.vector(new float[] {0.08f, -0.62f, 0.39f}));
    Optional<Document> result = collection.findOneAndUpdate(null, update, options);
    System.out.println(result);
  }
}

File: clients/csharp/examples/collections-find-one-and-update-vector.cs
using System.Text.Json;
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Collections;
using DataStax.AstraDB.DataApi.Core;
using DataStax.AstraDB.DataApi.Core.Query;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Update a document
    var embeddings = new float[] { 0.08f, -0.62f, 0.39f };
    var updateOptions = new CollectionFindOneAndUpdateOptions<Document>()
    {
      Sort = Builders<Document>.CollectionSort.Vector(embeddings),
    };
    var update = Builders<Document>.CollectionUpdate.Set("color", "blue");
    var result = await collection.FindOneAndUpdateAsync(
      null,
      update,
      updateOptions
    );

    Console.WriteLine(JsonSerializer.Serialize(result));
  }
}
